tests/sample_test/CMakeLists.txt

```diff
@@ -0,0 +1,17 @@
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.20.0)
+
+# create compile_commands.json for clang
+set(CMAKE_EXPORT_COMPILE_COMMANDS on)
+
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(sample_test)
+
+# Add source files with test code
+file(GLOB app_sources src/*.c)
+target_sources(app PRIVATE ${app_sources})
+
+# add test target All header and source files required for this test must be
+# added here target_include_directories(app PRIVATE ${APP_DIR}/some/header/path)
+# target_sources(app PRIVATE ${APP_DIR}/some/source/path)
```

Comment:
This comment is fine but could be better. We could describe here that some tests do not require anything, since they will use Kconfig to select the module they are testing, and thus "gain access" to the source and header file.

If, on the other hand, you are testing an internal header of some module/lib, or only a specific subset of source files from within some module/lib, you must add the headers and sources manually, as well as set any defines used in the module/lib. 

For example, this is how I configured a test for a "submodule" within an application:

```cmake
set(APP_DIR ../../app/main/src)

target_compile_definitions(app PRIVATE CONFIG_EXP_BACKOFF_MAX_NUM=2)
target_include_directories(app PRIVATE ${APP_DIR}/networking/conn_mgmt/exp_backoff)
target_sources(app PRIVATE ${APP_DIR}/networking/conn_mgmt/exp_backoff/exp_backoff.c)

```

Reply:
You can also add a `Kconfig` file to the test project and the resource some file from within you app. For example, instead of the `target_compile_definitions` above, I could have used
`rsource ../../app/main/src/networking/conn_mgmt/exp_backoff/Kconfig`
to get that Kconfig simbol and then set it in prj.conf